### PR TITLE
Generate CodeView by deafult for Windows

### PR DESCRIPTION
**Pull request by lanza**
llvm, lldb and swift combine to offer better support for PDBs on
Windows than it does DWARF. Default to generating PDBs on Windows.

### CODE REVIEW

**Comment by lanza** [comment]
After #22624 goes in this would be the most sane default.

**Comment by compnerd** [comment]
Please add a driver level test to ensure that we do this.

**Comment by compnerd** [comment]
CC: @dcci

**Comment by lanza** [comment]
@compnerd This fails to build on Windows due to the pdb gen bug you hit. WIP

**Comment by CodaFi** [comment]
Let's see what happens

@swift-ci please smoke test

**Comment by CodaFi** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
welp, thats going to destroy the test suite on Windows; I think that we should ensure that those work before merging this

**Review by dcci — CHANGES_REQUESTED** [review]
Agree. Also it feels like a bug to me that Windows failing doesn't prevent the PR from being merged.

**Comment by CodaFi** [comment]
@lanza I think we can merge this if you XFAIL or fix these tests:

```
    Swift(windows-x86_64) :: IRGen/condfail_message.swift
    Swift(windows-x86_64) :: DebugInfo/variables.swift
    Swift(windows-x86_64) :: DebugInfo/tuple.swift
    Swift(windows-x86_64) :: DebugInfo/top_level_code.swift
    Swift(windows-x86_64) :: DebugInfo/shadowcopy-linetable.swift
    Swift(windows-x86_64) :: DebugInfo/scopes.swift
    Swift(windows-x86_64) :: DebugInfo/return.swift
    Swift(windows-x86_64) :: DebugInfo/prologue.swift
    Swift(windows-x86_64) :: DebugInfo/patternmatching.swift
    Swift(windows-x86_64) :: DebugInfo/linetable.swift
    Swift(windows-x86_64) :: DebugInfo/line-directive.swift
    Swift(windows-x86_64) :: DebugInfo/letstring.swift
    Swift(windows-x86_64) :: DebugInfo/iteration.swift
    Swift(windows-x86_64) :: DebugInfo/inlinedAt.swift
    Swift(windows-x86_64) :: DebugInfo/inlinescopes.swift
    Swift(windows-x86_64) :: DebugInfo/implicitreturn.swift
    Swift(windows-x86_64) :: DebugInfo/gsil.swift
    Swift(windows-x86_64) :: DebugInfo/foreach.swift
    Swift(windows-x86_64) :: DebugInfo/fnptr.swift
    Swift(windows-x86_64) :: DebugInfo/enum.swift
    Swift(windows-x86_64) :: DebugInfo/columns.swift
    Swift(windows-x86_64) :: DebugInfo/closure-args.swift
    Swift(windows-x86_64) :: DebugInfo/basic.swift
    Swift(windows-x86_64) :: DebugInfo/bool.swift
    Swift(windows-x86_64) :: DebugInfo/any.swift
    Swift(windows-x86_64) :: DebugInfo/alloca-init.swift
    Swift(windows-x86_64) :: DebugInfo/NestedTypes.swift
    Swift(windows-x86_64) :: DebugInfo/Imports.swift
    Swift(windows-x86_64) :: DebugInfo/ImportsStdlib.swift
    Swift(windows-x86_64) :: DebugInfo/EagerTypeMetadata.swift
```

**Comment by dcci** [comment]
I wouldn't really XFAIL these tests. It means that there are potentially big holes in CodeView Windows support and merging this patch would lead to false expectations of what works and what doesn't (or maybe tests just need to be updated and everything works fine, in which case, well, we should do it sooner rather than later).

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by CodaFi** [comment]
@lanza I'm going to close this out since it seems like this needs to be handled incrementally to avoid disrupting the Windows build.

**Pull request closed without merging**
